buildpack/layermetadata/encoderdecoder.go

```diff
@@ -0,0 +1,129 @@
+package layermetadata
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/BurntSushi/toml"
+
+	"github.com/buildpacks/lifecycle/api"
+)
+
+type EncoderDecoder interface {
+	IsSupported(buildpackAPI string) bool
+	Encode(file *os.File, lmf File) error
+	Decode(path string) (File, string, error)
+}
+
+func supportedEncoderDecoders() []EncoderDecoder {
+	return []EncoderDecoder{
+		&DefaultEncoderDecoder{},
+		&LegacyEncoderDecoder{},
```

Comment:
```suggestion
		&defaultEncoderDecoder{},
		&legacyEncoderDecoder{},
```

This is definitely a nitpick but most of the methods, and even the interface, is not used outside of this package. Having _only_ `EncodeFile(), DecodeFile()` be given the public scope, and be moved to the top of the file, makes the file easier to parse. And better expresses developer intent.

Reply:
Please see 418562fc